src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1142,6 +541,15 @@ protected virtual void SetupArgumentsForElementMarshalling()
             _nativeHome = new Home(emitter.NewLocal(NativeType), NativeType, isByRef: false);
         }
 
+        protected virtual void SetupArgumentsForFieldMarshalling()
+        {
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+            ILCodeStream marshallingCodeStream = _ilCodeStreams.MarshallingCodeStream;
+
+            _managedHome = new Home(emitter.NewLocal(ManagedType), ManagedType, isByRef: false);
```

Comment:
Some comments please. They are more like temporary locals used for propagating values.